Doc/reference/datamodel.rst

```diff
@@ -2144,8 +2144,10 @@ through the container; for mappings, :meth:`__iter__` should be the same as
 
    Called to implement :func:`operator.length_hint`. Should return an estimated
    length for the object (which may be greater or less than the actual length).
-   The length must be an integer ``>=`` 0. This method is purely an
-   optimization and is never required for correctness.
+   The length must be an integer ``>=`` 0. The return value may also be
+   *NotImplemented*, which is treated the same as if the ``__length_hint__``
```

Comment:
```suggestion
   :const:`NotImplemented`, which is treated the same as if the ``__length_hint__``
```
Let's add a Sphinx role so that users can click on the constant to be directed towards its documentation. This change will roughly convert *NotImplemented* -> [NotImplemented](https://docs.python.org/3/library/constants.html#NotImplemented). It might be helpful for users who aren't familiar with it.

This of course doesn't need be done for ``__length_hint__`` since this is within its dedicated section.